content/en/real_user_monitoring/browser/monitoring_page_performance.md

```diff
@@ -147,6 +147,49 @@ The RUM SDK automatically monitors frameworks that rely on hash (`#`) navigation
 
 ## Add your own performance timing
 
+### Custom Vitals
+
+The `customVital` API allows you to measure the performance of your application at the component level. You can use custom vitals to measure how quickly part of your page renders, or how long it takes for a component to respond to a user interaction.
+
+#### startDurationVital and stopDurationVital
+
+You can start a duration measurement by calling `startDurationVital` and stopping it later in your code with `stopDurationVital`:
+
+```javascript
+window.DD_RUM.startDurationVital("dropdownRendering")
+window.DD_RUM.stopDurationVital("dropdownRendering")
+```
+
+Once you call the `stopDurationVital` method, the custom vital duration is sent to Datadog and can be queried using `@vital.name:dropdownRendering`. Let's use `@vital.duration:>10` to demonstrate how to filter for long durations.
+
+#### reference and description
+
+Additionally, you can use the reference returned by `startDurationVital` and specify a `description` string to differentiate between different instances of the same custom vital, like when a component is used on multiple pages. For example, to track the duration of `dropdownRendering` in the `login` page:
+
+```javascript
+const reference = window.DD_RUM.startDurationVital("dropdownRendering", { description: "login" })
+window.DD_RUM.stopDurationVital(reference)
+```
+
+This allows you to view the same component's rendering behavior across different pages by grouping by `@vital.description`.
+
+You can also add additional context to your custom vital by using the `context` property:
+
+```javascript
+window.DD_RUM.startDurationVital("dropdownRendering", {context: { clientId: "xxx" }})
+window.DD_RUM.stopDurationVital("dropdownRendering")
+```
+
+#### addDurationVital
+
+Alternatively, you can report a custom vital in a single operation using `addDurationVital`:
```

Comment:
```suggestion
Instead of setting custom vital variables individually, you can report a custom vital in a single operation using `addDurationVital`:
```